### PR TITLE
feat(buttons): add support for checkbox buttons

### LINKED ISSUE #890: Button groups does not support focus customization
### Bug description:

Button groups do not support styling the `:focus` state
### Link to minimally-working plunker that reproduces the issue:

https://ng-bootstrap.github.io/#/components/buttons

Click on any "button" and press tab, the next button show no focus decoration
### Root cause

Compare the ng-bs markup:

``` html
<div class="btn-group" data-toggle="buttons">
  <label class="btn btn-primary" [class.active]="model.left">
    <input type="checkbox" [(ngModel)]="model.left"> Left (pre-checked)
  </label>
</div>
```

with the [bs4 markup](http://v4-alpha.getbootstrap.com/components/button-group/)

``` html
<div class="btn-group" role="group" aria-label="Basic example">
  <button type="button" class="btn btn-secondary">Left</button>
</div>
```

bs4  uses the focus style of the `button` - I'm not sure there is anything we can do with the current example in ngbs ?
### Version of Angular, ng-bootstrap, and Bootstrap:

ng: 2.0.0
bs: 4.0.0-alpha.4
ng-bs: 1.0.0-alpha.7

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Closes #890

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@maxokorokov could you PTAL?

File: src/buttons/checkbox.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    '(focus)': 'focused = true',
+    '(blur)': 'focused = false'
+  },
+  providers: [NGB_CHECKBOX_VALUE_ACCESSOR]
+})
+export class NgbCheckBox implements ControlValueAccessor {
+  checked;
+
+  /**
+   * A flag indicating if a given radio button is disabled.
```
Comment (maxokorokov) [inline src/buttons/checkbox.ts]: radio button → checkbox

File: src/buttons/label.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import {Directive, Renderer2, ElementRef} from '@angular/core';
+
+@Directive({selector: '[ngbButtonLabel]', host: {'[class.btn]': 'true'}})
+export class NgbButtonLabel {
+  constructor(private _renderer: Renderer2, private _elRef: ElementRef) {}
```
Comment (maxokorokov) [inline src/buttons/label.ts]: True that we could have done it with 3 fields and host bindings, don't really need Renderer, right?

File: src/buttons/checkbox.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  multi: true
+};
+
+
+/**
+ * Easily create Bootstrap-style checkbox buttons. A value of a checked button is bound to a variable
+ * specified via ngModel.
+ */
+@Directive({
+  selector: '[ngbButton][type=checkbox]',
```
Comment (maxokorokov) [inline src/buttons/checkbox.ts]: Ha, that's a much better solution!

File: src/index.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ export {
   NgbPanelContent
 } from './accordion/accordion.module';
 export {NgbAlertModule, NgbAlertConfig, NgbAlert} from './alert/alert.module';
-export {NgbButtonsModule, NgbRadioGroup} from './buttons/radio.module';
+export {NgbButtonsModule, NgbRadioGroup} from './buttons/buttons.module';
```
Comment (maxokorokov) [inline src/index.ts]: Seems strange, that we have only `NgbRadioGroup` here

File: src/buttons/label.ts (hunk 2 above)
Comment (pkozlowski-opensource) [inline src/buttons/label.ts]: Yep, will send a follow up PR to refactor it. Just didn't want to make too many changes in one PR.

File: src/index.ts (hunk 4 above)
Comment (pkozlowski-opensource) [inline src/index.ts]: This is so people can query for it. We could add the new checkbox button here as well.

**Pull request closed without merging**